scripts/inactive_study.py

```diff
@@ -162,3 +174,60 @@
     plt.legend(loc='lower right')
     plt.savefig('inactive_study/' + args.script_name + '_k' + str(nl) + '.pdf', bbox_inches='tight')
     plt.close()
+
+# loop over each layer (index i) and report inactive batch (index j) that satisfies
+# convergence criteria for the selected method (window,half, or all)
+for i in range(len(n_layers)):
+  nl = n_layers[i]
+  if(method == "window"):
+      # detect when entropy of current batch is within window (mean +/- std of window)
+      for j in range(window_length,len(entropy[i])):
+          window = entropy[i][(j-window_length):j]
+          window_mean = np.average(window)
+          window_dev = np.std(window)
+          window_low = window_mean - window_dev
+          window_high = window_mean + window_dev
+          # print("window bounds: ", window_low , " to ", window_high )
+          if(window_high - entropy[0][j] > 0 and entropy[0][j]-window_low > 0):
+              # if entropy[j] is less than window_high and greater than
+              # window_low, then it is within the window and we've succeeded
+              print("For layer", nl, ", batch", j, "produced a value within one "
+                    "standard deviation of the window mean. It is recommended to do "
+                    "at least this many inactive cycles for this layer.")
+              break
+          else:
+              continue
+  elif(method == "half"):
+      #  Accordinig to Brown, the below is done in MCNP5:
+      # find the first cycle when Hsrc is within one standard deviation of its average for the last half of cycles
+      # compute on the fly the average of the eigenvalue (last half batches) +/- the std dev of the last half cycles
+      # and see if the current k is within this interval
+      for j in range(1,len(entropy[i])):
+          last_half_idx = int(np.floor(j/2))
+          last_half_vals = entropy[0][last_half_idx:j]
+          stdev = np.std(last_half_vals)
+          mean = np.mean(last_half_vals)
+          low =  mean - stdev
+          high = mean + stdev
+          if(high - entropy[0][j] > 0 and entropy[0][j]-low > 0):
+              print("For layer", nl, ", batch", j, "produced a value within one "
+                    "standard deviation of the mean for the last half of entropy values. "
+                    "It is recommended to do at least this many inactive cycles for this layer.")
+              break
+          else:
+              continue
+  else:
+      # use all batches as data points
+      for j in range(1,len(entropy[i])):
+          entropy_slice = entropy[i][0:j]
+          stdev = np.std(entropy_slice)
+          mean = np.mean(entropy_slice)
+          low =  mean - stdev
+          high = mean + stdev
+          if(high - entropy[i][j] > 0 and entropy[i][j]-low > 0):
+              print("For layer", nl, ", batch", j, "produced a value within one "
```

Comment:
I am wondering how these multiple print statements look when shown to the screen. If the user has, say, 8 different choices for layers, this might get quite busy. What do you think about showing the same info, but in a table? 

I'd also suggest showing some information telling the user if they didn't run enough inactive batches at all to meet the criteria. I think right now, that currently nothing will print out at all (which would be confusing).

Reply:
Any other places in Cardinal that make a table the way you want I could copy / things to `grep` for?

I can add a message for the case that convergence is not detected with the user settings.

Reply:
Nope, go ahead and just make a table you think looks nice

Reply:
I've been looking into python packages that automate table formatting, but it seems they all require a pip install. Do you have reservations about relying on something like this or do you think it is fine for the user to run the script, see they are missing the package and then figure out they need to install it? I can put some kind of note in the script that just tells them to pip install as well for extra support.